src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/Int32/SparseRangeInt32FrozenSet.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NET7_0_OR_GREATER
+
+using System.Collections.Generic;
+
+namespace System.Collections.Frozen
+{
+    /// <summary>Provides a frozen set to use when the value is an <see cref="int"/>, the default comparer is used, and the items are not in a contiguous range.</summary>
+    /// <remarks>
+    /// No hashing here, just a direct lookup into a bit vector.
+    /// </remarks>
+    internal sealed class SparseRangeInt32FrozenSet : FrozenSetInternalBase<int, SparseRangeInt32FrozenSet.GSW>
```

Comment:
Can you share a couple of real examples where this will end up being used?  As with my earlier comment/question, I'm fine with it if it makes a meaningful difference in important cases... I just want to avoid adding a bunch of code that optimizes a case just because we can.